enigma-principal/app/src/cli/app.rs

```diff
@@ -1,32 +1,67 @@
 use boot_network::{
     keys_provider_http::{PrincipalHttpServer, StateKeyRequest},
-    principal_manager::{self, PrincipalManager, ReportManager, Sampler, PrincipalConfig},
+    principal_manager::{self,
+                        PrincipalManager,
+                        ReportManager,
+                        Sampler,
+                        PrincipalConfig,
+                        SgxEthereumSigner,
+                        PrivateKeyEthereumSigner},
 };
 use cli;
+use enigma_crypto::EcdsaSign;
 use enigma_tools_u::{esgx::general::storage_dir, web3_utils::enigma_contract::EnigmaContract};
 use epoch_u::epoch_provider::EpochProvider;
 use esgx::general::ENCLAVE_DIR;
 use failure::Error;
 use sgx_types::sgx_enclave_id_t;
 use std::{fs::File, io::prelude::*, path::Path, sync::Arc};
 use structopt::StructOpt;
+use rustc_hex::FromHex;
+
+pub fn create_signer(eid: sgx_enclave_id_t, with_private_key: bool, private_key: Vec<u8>) -> Box<dyn EcdsaSign + Send + Sync> {
+    if with_private_key {
+        let mut pk_32 = [0u8; 32];
+        for (i, v) in private_key.iter().enumerate() { pk_32[i] = *v; }
```

Comment:
instead of doing this, set the type of `private_key` in the fn signature to `&[u8]` and then use `pk_32.copy_from_slice(private_key)`
HINT: when calling this function, just add a `&` before the `private_key`. it will do the right conversion using `Deref`

Reply:
Generally i don't love this function's signature but we can leave that for future refactoring.

Reply:
> instead of doing this, set the type of `private_key` in the fn signature to `&[u8]` and then use `pk_32.copy_from_slice(private_key)`
> HINT: when calling this function, just add a `&` before the `private_key`. it will do the right conversion using `Deref`

Works for me also without changing the fn signature. Sounds okay?